Tests/GRDBTests/FetchableRecordDecodableTests.swift

```diff
@@ -301,3 +301,197 @@ extension FetchableRecordDecodableTests {
         XCTAssertEqual(value.uuid, uuid)
     }
 }
+
+// MARK: - Custom nested Decodable types - nested saved as JSON
+
+extension FetchableRecordDecodableTests {
+    func testOptionalNestedStruct() throws {
+        struct NestedStruct : PersistableRecord, FetchableRecord, Codable {
+            let firstName = "Bob"
+            let lastName = "Dylan"
```

Comment:
I'm glad a little music is entering the tests 👍

Reply:
BTW... When I read `let firstName = "Bob"`, I wonder how NestedStruct could hold different values. This makes the  `XCTAssertEqual(nestedModel.firstName, "Bob")` test less obvious.

I'm not saying there is a bug here. I'm saying the test would be clearer if NestedStruct was declared as:

```swift
struct NestedStruct : PersistableRecord, FetchableRecord, Codable {
    var firstName: String
    var lastName: String
}
```

and the insertion below written as:

```swift
let value = StructWithNestedType(nested: NestedStruct(firstName: "Bob", lastName: "Dylan"))
```

This comment applies to other testing structs with `let` properties.

Reply:
BTW... When I read `let firstName = "Bob"`, I wonder how NestedStruct could hold different values. This makes the  `XCTAssertEqual(nestedModel.firstName, "Bob")` test less obvious.

I'm not saying there is a bug here. I'm saying the test would be clearer if NestedStruct was declared as:

```swift
struct NestedStruct : PersistableRecord, FetchableRecord, Codable {
    var firstName: String
    var lastName: String
}
```

and the insertion below written as:

```swift
let value = StructWithNestedType(nested: NestedStruct(firstName: "Bob", lastName: "Dylan"))
```

This comment applies to other testing structs with `let` properties.

Reply:
Another point: NestedStruct does not need to adopt PersistableRecord and FetchableRecord. Same for all other nested structs.